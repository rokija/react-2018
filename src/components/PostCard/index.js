import React from "react";
import "./index.css";

export const sum = (a, b) => {
  return a + b;
};

const PostCard = ({ image, title, text }) => (
  <div className="col-12">
    <div className="card">
      <img className="card-img-top" src={image} />
      <div className="card-body">
        <h1 className="card-title">{title}</h1>
        <span className="card-text">{text}</span>
      </div>
    </div>
  </div>
);

export default PostCard;
