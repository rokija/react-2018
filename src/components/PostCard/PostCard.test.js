import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import PostCard, { sum } from "./index";

describe("testing PostCard", () => {
  const wrapper = mount(<PostCard />);

  it("sums two numbers", () => {
    expect(sum(3, 5)).toEqual(8);
  });

  it("renders properly", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("renders a snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
