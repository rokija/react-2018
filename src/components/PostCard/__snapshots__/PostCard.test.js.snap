// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing PostCard renders a snapshot 1`] = `
<PostCard>
  <div
    className="col-12"
  >
    <div
      className="card"
    >
      <img
        className="card-img-top"
      />
      <div
        className="card-body"
      >
        <h1
          className="card-title"
        />
        <span
          className="card-text"
        />
      </div>
    </div>
  </div>
</PostCard>
`;
